src/Sulu/Component/Content/Document/Subscriber/WorkflowStageSubscriber.php

```diff
@@ -12,85 +12,86 @@
 namespace Sulu\Component\Content\Document\Subscriber;
 
 use PHPCR\NodeInterface;
-use PHPCR\PropertyType;
 use Sulu\Component\Content\Document\Behavior\WorkflowStageBehavior;
 use Sulu\Component\Content\Document\WorkflowStage;
-use Sulu\Component\DocumentManager\DocumentAccessor;
-use Sulu\Component\DocumentManager\Event\AbstractMappingEvent;
-use Sulu\Component\DocumentManager\Event\HydrateEvent;
+use Sulu\Component\DocumentManager\Event\MetadataLoadEvent;
 use Sulu\Component\DocumentManager\Event\PersistEvent;
+use Sulu\Component\DocumentManager\Events;
+use Sulu\Component\DocumentManager\PropertyEncoder;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
 
-class WorkflowStageSubscriber extends AbstractMappingSubscriber
+class WorkflowStageSubscriber implements EventSubscriberInterface
 {
     const WORKFLOW_STAGE_FIELD = 'state';
     const PUBLISHED_FIELD = 'published';
 
-    public function supports($document)
-    {
-        return $document instanceof WorkflowStageBehavior;
-    }
+    /**
+     * @var PropertyEncoder
+     */
+    private $encoder;
 
     /**
-     * @param HydrateEvent $event
+     * @param PropertyEncoder $encoder
      */
-    public function doHydrate(AbstractMappingEvent $event)
+    public function __construct(
+        PropertyEncoder $encoder
+    ) {
+        $this->encoder = $encoder;
+    }
+
+    public static function getSubscribedEvents()
     {
-        $locale = $event->getLocale();
-        $node = $event->getNode();
-        $document = $event->getDocument();
+        return [
+            Events::METADATA_LOAD => 'handleMetadataLoad',
+            Events::PERSIST => 'handlePersist',
+        ];
+    }
 
-        $workflowStage = $this->getWorkflowStage($node, $locale);
-        $document->setWorkflowStage($workflowStage);
+    public function handleMetadataLoad(MetadataLoadEvent $event)
+    {
+        $metadata = $event->getMetadata();
 
-        $publishedDate = $event->getNode()->getPropertyValueWithDefault(
-            $this->encoder->localizedSystemName(self::PUBLISHED_FIELD, $event->getLocale()),
-            null
-        );
-        $event->getAccessor()->set(
-            'published',
-            $publishedDate
-        );
+        if (false === $metadata->getReflectionClass()->isSubclassOf(WorkflowStageBehavior::class)) {
+            return;
+        }
+
+        $metadata->addFieldMapping('workflowStage', [
+            'encoding' => 'system_localized',
+            'property' => self::WORKFLOW_STAGE_FIELD,
+            'type' => 'long',
+        ]);
+        $metadata->addFieldMapping('published', [
+            'encoding' => 'system_localized',
+            'property' => self::PUBLISHED_FIELD,
+            'type' => 'date',
+        ]);
     }
 
     /**
      * @param PersistEvent $event
      */
-    public function doPersist(PersistEvent $event)
+    public function handlePersist(PersistEvent $event)
```

Comment:
Ok, so this can only be omitted if the value of the node and document are exactly the same?